From 37de1aaae1edd1fbcd3c2ef2ac5f4baa4c8e2751 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Tue, 11 Jan 2022 18:14:35 -0500
Subject: [PATCH] Revert "build: disable gir install via list to pacify meson
 >= 0.60.2 (#10489)"

This reverts commit 8fc2df08b40aa3e1958ed2fde853c50676d8cf48.

This commit was wrong, because it tried to work around a bug in a single
version of meson by using something that isn't, wasn't, and won't be a
valid value.

The fixed version of meson 0.60.x has been out for a while now, which
once again accepts `false`, and 0.61.0 also accepts `false` but was
known at the time of this workaround to not work in meson-git master
(now meson 0.61.0).

Using `false` is acceptable and the failure to accept it has been
qualified as a meson regression. Using `[false]` is just... trying to
fuzz meson with random objects until you get something that slips its
way through the argument checker and produces desired effects on the
python implementation level.
---
 src/meson.build    | 2 +-
 src/st/meson.build | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index 7999c0a67b..34b130d50f 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -196,7 +196,7 @@ cinnamon_gir = gnome.generate_gir(
     includes: cinnamon_gir_includes,
     install: true,
     install_dir_typelib: pkglibdir,
-    install_dir_gir: [false],
+    install_dir_gir: false,
     extra_args: [
         '-DST_COMPILATION',
         '--quiet',
diff --git a/src/st/meson.build b/src/st/meson.build
index d299727d06..ec7d7b30a1 100644
--- a/src/st/meson.build
+++ b/src/st/meson.build
@@ -213,7 +213,7 @@ st_gir = gnome.generate_gir(
     includes: st_gir_includes,
     install: true,
     install_dir_typelib: pkglibdir,
-    install_dir_gir: [false],
+    install_dir_gir: false,
     extra_args: [
         '-DST_COMPILATION',
         '--quiet',
